src/main/java/org/opensearch/timeseries/util/ParseUtils.java

```diff
@@ -473,6 +474,39 @@ public static SearchSourceBuilder addUserBackendRolesFilter(User user, SearchSou
         return searchSourceBuilder;
     }
 
+    public static void addAccessibleConfigsFilterAndSearch(
+        Client client,
+        Pair<String, String> pair,
+        SearchRequest searchRequest,
+        ActionListener<SearchResponse> listener
+    ) {
+        logger.debug("Filtering result by accessible resources");
+        ResourceSharingClient resourceSharingClient = ResourceSharingClientAccessor.getInstance().getResourceSharingClient();
+        SearchSourceBuilder searchSourceBuilder = searchRequest.source();
+        resourceSharingClient.getAccessibleResourceIds(pair.getLeft(), ActionListener.wrap(configIds -> {
```

Comment:
makes as call to get accessible Id and then adds a filter to current search query to only search from those ids.